provider/datagen/src/transform/cldr/source.rs

```diff
@@ -9,71 +9,51 @@ use std::fmt::Debug;
 use std::str::FromStr;
 
 /// Specifies a variant of CLDR JSON
-#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq, serde::Deserialize)]
 #[non_exhaustive]
-pub enum LocaleSubset {
```

Comment:
Nit (optional): It doesn't _hurt_ to keep the LocaleSubset stuff around; it results in slightly smaller file downloads. I have a small preference to keep LocaleSubset working the way it's always been working (with the `--cldr-locale-subsets` CLI flag from 1.0), and just layer this on top of it.

Reply:
We agreed to always download full as then we benefit from caching.